packages/core/services/jwk.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('fs');
+const path = require('path');
+const { generateKeyPair, exportJWK, importJWK, jwtVerify } = require('jose');
+
+// See https://github.com/CommunitySolidServer/CommunitySolidServer/blob/15a929a87e4ce00c0ed266e296405c8e4a22d4a7/src/identity/configuration/CachedJwkGenerator.ts
+module.exports = {
+  name: 'jwk',
+  settings: {
+    jwtPath: null,
+    alg: 'ES256'
+  },
+  async created() {
+    const privateKeyPath = path.resolve(this.settings.jwtPath, 'jwk' + this.settings.alg + '.key');
+    const publicKeyPath = path.resolve(this.settings.jwtPath, 'jwk' + this.settings.alg + '.key.pub');
+
+    if (!fs.existsSync(privateKeyPath) && !fs.existsSync(publicKeyPath)) {
+      console.log('JWK not found, generating...');
+      if (!fs.existsSync(this.settings.jwtPath)) {
+        fs.mkdirSync(this.settings.jwtPath);
+      }
+      await this.actions.generateKeyPair({ privateKeyPath, publicKeyPath });
+    } else {
+      this.privateJwk = JSON.parse(fs.readFileSync(privateKeyPath));
+      this.publicJwk = JSON.parse(fs.readFileSync(publicKeyPath));
+    }
+  },
```

Comment:
Why are the keys stored on the filesystem and not the db?

Reply:
That's what we do with the similar `auth.jwt` service. But indeed it could be saved in the `account` dataset